### PR TITLE
Added the function `generate_id` to create a unique UUID. 

### PR DESCRIPTION
**Pull request by rafgia**
This pull request:
    - Implements function to generate a universal unique identifier (UUID) for each row.
    - Modifies the target column in the DataFrame with generated UUIDs.
    - Ensures that UUIDs are correctly applied for each row based on the source column.
    - Add a test for the `generate_id` function.

### CODE REVIEW

File: omopetl/transform.py [hunk 1]
```diff
@@ -72,6 +84,23 @@ def transform_normalize_date(self, source_column, target_column, transformation)
...
+        To generate a universal unique identifier for each patient
+        
+        """
+        generated_uuid = uuid.uuid4()
+        return farmhash.fingerprint64(str(generated_uuid))
+
+    def coalesce(*args):
+        """
+        
+        Return 0 if the result is NULL
```
Comment (tompollard) [inline omopetl/transform.py:99]: `Return 0 if the result is NULL` doesn't really describe the behaviour of the function. Instead how about something like: `Return the first non-null value from a list of arguments.`

**Review by tompollard — COMMENTED** [review]
Thanks for looking at this @rafgia. Tests are failing:

```
==================================== ERRORS ====================================
__________________ ERROR collecting tests/test_transformer.py __________________
ImportError while importing test module '/home/runner/work/omopetl/omopetl/tests/test_transformer.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_transformer.py:3: in <module>
    from omopetl.transform import Transformer
omopetl/transform.py:3: in <module>
    import farmhash
E   ModuleNotFoundError: No module named 'farmhash'
```

Looks like farmhash needs to be added to the requirements file. 

Do you more details about the package and why you chose it for the hashes? Why not use `import uuid` which (I think) is part of Python's standard library, so avoids the extra dependency.

**Comment by rafgia** [comment]
In the `mappings.yaml` file, for the person mapping, we have `source_column: dob`, but the `patients` table does not contain this column.

**Comment by tompollard** [comment]
> In the mappings.yaml file, for the person mapping, we have source_column: dob, but the patients table does not contain this column.

Yes, sorry, I am working on this now. I updated the source CSV files, but I haven't yet written the mappings. I will open a new PR with the fix in 30 minutes or so...

**Comment by tompollard** [comment]
@rafgia, the following issue should now be resolved:

> In the mappings.yaml file, for the person mapping, we have source_column: dob, but the patients table does not contain this column.

You will need to rebase on main to pull the fix into your branch.

**Comment by tompollard** [comment]
@rafgia do you want to drop over to my office to sort this out?

File: omopetl/transform.py [hunk 2]
```diff
@@ -61,6 +63,11 @@ def transform_map(self, source_column, target_column, transformation):
         """Map values in the source column to new values."""
         value_map = transformation["values"]
         return self.data[source_column].map(value_map).astype("Int64")
+    
+    def transform_generate_id(self, source_column, target_column, transformation):
+        """Generate a universal unique identifier for each row in the source column."""
+        self.data[target_column] = [uuid.uuid4() for _ in range(len(self.data[source_column]))]
```
Comment (tompollard) [inline omopetl/transform.py:69]: `self.data` is the source data. I don't think we want to overwrite here?

File: omopetl/transform.py [hunk 3]
```diff
@@ -83,6 +89,26 @@ def transform_normalize_date(self, source_column, target_column, transformation)
         date_format = transformation.get("format", "%Y-%m-%d")
         return pd.to_datetime(self.data[source_column], errors="coerce").dt.strftime(date_format)
 
+    def transform_filter(self, source_column, target_column, transformation):
```
Comment (tompollard) [inline omopetl/transform.py:92]: This function was removed in https://github.com/tompollard/omopetl/pull/21. It looks like an outdated main branch has been merged into this branch.

**Comment by rafgia** [comment]
@tompollard I've cleaned up the code and rebased on main.

**Pull request merged**